psh/mkdir/mkdir.c

```diff
@@ -35,8 +35,13 @@ int psh_mkdir(int argc, char **argv)
 	}
 
 	for (i = 1; i < argc; i++) {
-		if (mkdir(argv[i], 0) < 0)
-			fprintf(stderr, "mkdir: failed to create %s directory\n", argv[i]);
+		if (mkdir(argv[i], 0) < 0) {
+			fprintf(stderr, "mkdir: failed to create %s directory", argv[i]);
```

Comment:
I would prefer to use `strerror` to handle all errors - not just "name being too long" special case.

Please note that for common errors (no extra arguments in format) `perror` can be used as an even shorter form.

Reply:
You are right, streeror() seems to be a better solution